### PR TITLE
[Telink] Introduce power management

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

- Add power management (disabled by default for all examples)
- Rework Factory Reset button from "Hold" (3 s) into "Push" (3 times)
- Add BUTTON_MANAGER_IRQ_MODE feature
- Add CHIP_ENABLE_APPLICATION_STATUS_LED feature

**Testing**

Tested manually with chip-tool.

Steps:

- Run: $ chip-tool pairing ble-thread <...>
- Wait till success
- Run: $ chip-tool onoff on ${NODE_ID }1
- Wait till success
- Run: $ chip-tool onoff off ${NODE_ID }1
- Wait till success

### CODE REVIEW

File: src/platform/telink/CHIPDevicePlatformConfig.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  *
- *    Copyright (c) 2022 Project CHIP Authors
+ *    Copyright (c) 2023 Project CHIP Authors
```
Comment (bzbarsky-apple) [inline src/platform/telink/CHIPDevicePlatformConfig.h:3]:
```suggestion
 *    Copyright (c) 2022-2023 Project CHIP Authors
```
?

Comment (s07641069) [reply src/platform/telink/CHIPDevicePlatformConfig.h]: Fixed for all files

File: examples/all-clusters-app/telink/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-#    Copyright (c) 2022 Project CHIP Authors
+#    Copyright (c) 2022-2023 Project CHIP Authors
```
Comment (andy31415) [inline examples/all-clusters-app/telink/CMakeLists.txt:2]: These changes are not necessary: files are completely unchanged. This just creates change logs.

File: examples/all-clusters-app/telink/include/AppTask.h [hunk 3]
```diff
@@ -70,6 +79,11 @@ class AppTask
 
     static AppTask sAppTask;
     PWMDevice mPwmIdentifyLed;
+
+#if CONFIG_CHIP_FACTORY_DATA
+    // chip::DeviceLayer::FactoryDataProvider<chip::DeviceLayer::InternalFlashFactoryData> mFactoryDataProvider;
```
Comment (andy31415) [inline examples/all-clusters-app/telink/include/AppTask.h:84]: Do not check in commented-out code.
If you want to also menction `InternalFlashFactoryData` exists, then comment stating that, don't put a line of code without context.

File: examples/all-clusters-app/telink/src/AppTask.cpp [hunk 4]
```diff
@@ -163,10 +167,30 @@ CHIP_ERROR AppTask::Init(void)
...
+        LOG_ERR("mFactoryDataProvider.GetEnableKey() failed. Could not delegate a test event trigger");
+        memset(sTestEventTriggerEnableKey, 0, sizeof(sTestEventTriggerEnableKey));
+    }
+#else
     SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+#endif
 
-    gExampleDeviceInfoProvider.SetStorageDelegate(&chip::Server::GetInstance().GetPersistentStorage());
+    static CommonCaseDeviceServerInitParams initParams;
+    // static OTATestEventTriggerDelegate testEventTriggerDelegate{ ByteSpan(sTestEventTriggerEnableKey) };
```
Comment (andy31415) [inline examples/all-clusters-app/telink/src/AppTask.cpp:189]: delete commented out code

**Review by andy31415 — APPROVED** [review]

File: examples/all-clusters-app/telink/CMakeLists.txt (hunk 2 above)
Comment (s07641069) [reply examples/all-clusters-app/telink/CMakeLists.txt]: Fixed

File: examples/all-clusters-app/telink/include/AppTask.h (hunk 3 above)
Comment (s07641069) [reply examples/all-clusters-app/telink/include/AppTask.h]: Fixed

**Review by Damian-Nordic — APPROVED** [review]

**Comment by s07641069** [comment]
@andy31415 @Damian-Nordic Could you process merge of PR if everything ok now?

**Pull request merged**